### ISSUE TITLE
Much lower mAP with same settings

### ISSUE THREAD

**Issue by NaomiEX** [HorizonRobotics/Sparse4D#14]
Hi, I trained the model with total batch size 48 and lr 6e-4 for 100 epochs, however I obtained a final mAP of only 44.1 which is much lower than the reported mAP. I've attached my log, let me know if you have any suggestions on why this could be. 

I've also seen the issue about Metric Fluctuations however the reported average there was 46.32 mAP which is still much higher than what I got.

[s4d.log](https://github.com/HorizonRobotics/Sparse4D/files/14318556/s4d.log)

**Comment by linxuewu** [comment]
Please follow the steps in [quick_start.md](https://github.com/HorizonRobotics/Sparse4D/blob/main/docs/quick_start.md) thoroughly, paying particular attention to compiling the 'deformable_aggregation' cuda operation. Your current GPU memory and training time are abnormal.

**Comment by NaomiEX** [comment]
I did follow the steps, I have compiled the cuda ops, the memory and training time difference is likely because I'm not training on RTX3090s but on 4 A100s

**Comment by linxuewu** [comment]
It's difficult to explain this reason. It might be due to batch normalization. You could try again with 8 GPUs or adjust the learning rate.

**Comment by linxuewu** [comment]
Pull the latest code and try again. @NaomiEX

**Comment by NaomiEX** [comment]
Alright, let me try again

**Comment by NaomiEX** [comment]
Apologies, I haven't been able to retrain it again, will report back once I train it with the new code when I get the time.

**Comment by linxuewu** [comment]
NDS is a more stable metric, prioritize NDS. Fluctuations don't appear significant from NDS.
These fluctuations could possibly be caused by resuming; completing a full training cycle might normalize the metrics.
@zhaoyangwei123

**Comment by tiger-sheep** [comment]
Hello, I also met the similar problem. I **evaluated** your published weight file "sparse4dv3_r50.pth" on 2*TiTAN RTX and  1*TiTAN RTX separately. And I obtained results as follows, which is much lower than reported metrics. I think that the evaluation result may be relative to the number of gpus? Thank you for your reply in advance!

**_2* TiTAN_**
mAP: 0.4652
mATE: 0.5401
mASE: 0.2618
mAOE: 0.4599
mAVE: 0.2199
mAAE: 0.2060
NDS: 0.5638
Eval time: 140.3s

Per-class results:
Object Class    AP      ATE     ASE     AOE     AVE     AAE
car     0.668   0.357   0.142   0.054   0.185   0.194
truck   0.395   0.527   0.187   0.052   0.163   0.211
bus     0.452   0.680   0.196   0.070   0.382   0.243
trailer 0.183   0.968   0.246   0.624   0.178   0.206
construction_vehicle    0.124   0.884   0.493   1.203   0.132   0.402
pedestrian      0.560   0.516   0.287   0.508   0.282   0.150
motorcycle      0.499   0.458   0.237   0.547   0.292   0.235
bicycle 0.425   0.439   0.256   0.960   0.144   0.006
traffic_cone    0.697   0.276   0.299   nan     nan     nan
barrier 0.650   0.294   0.276   0.123   nan     nan

**_1*TiTAN_**
mAP: 0.4649
mATE: 0.5416
mASE: 0.2623
mAOE: 0.4608
mAVE: 0.2198
mAAE: 0.2047
NDS: 0.5635
Eval time: 141.3s

Per-class results:
Object Class    AP      ATE     ASE     AOE     AVE     AAE
car     0.668   0.357   0.142   0.053   0.185   0.195
truck   0.394   0.527   0.187   0.051   0.163   0.211
bus     0.452   0.680   0.195   0.071   0.383   0.243
trailer 0.184   0.980   0.246   0.628   0.177   0.208
construction_vehicle    0.122   0.886   0.497   1.217   0.136   0.395
pedestrian      0.561   0.515   0.288   0.508   0.280   0.150
motorcycle      0.495   0.464   0.236   0.544   0.293   0.231
bicycle 0.428   0.438   0.257   0.953   0.141   0.004
traffic_cone    0.695   0.277   0.300   nan     nan     nan
barrier 0.650   0.292   0.275   0.121   nan     nan
